src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitSchedulerServiceTest.kt

```diff
@@ -202,5 +217,177 @@ class VisitSchedulerServiceTest {
           LocalDateTime.parse("2021-02-01T11:30")
         )
     }
+
+    @Test
+    fun `all sessions are returned when an offender has no non-associations`() {
+      val prisonId = "MDI"
+      val prisonerId = "A1234AA"
+      val startDate = LocalDate.parse("2021-02-01")
+
+      val singleSession = sessionTemplate(
+        startDate = startDate,
+        startTime = LocalTime.parse("11:30"),
+        endTime = LocalTime.parse("12:30"),
+        frequency = SessionFrequency.SINGLE
+      )
+      mockRepositoryResponse(listOf(singleSession))
+
+      Mockito.`when`(
```

Comment:
kotlin mockito provides `whenever` to avoid the clash with `when`

Reply:
Updated